optuna/samplers/_search_space.py

```diff
@@ -0,0 +1,111 @@
+from collections import OrderedDict
```

Comment:
Since `optuna/samplers/__init__.py` seemed to be getting bigger, I created the new module and moved the existing `intersection_search_space` function into this module.